hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyDeleteResponseV1.java

```diff
@@ -79,8 +81,18 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
           batchOperation, ozoneDbKey, omKeyInfo);
     } else {
       Table<String, OmKeyInfo> keyTable = omMetadataManager.getKeyTable();
+      OmKeyInfo omKeyInfo = getOmKeyInfo();
+      // Sets full absolute key name to OmKeyInfo, which is
+      // required for moving the sub-files to KeyDeletionService.
+      omKeyInfo.setKeyName(keyName);
+      String deletedKey = omMetadataManager
+          .getOzoneKey(omKeyInfo.getVolumeName(), omKeyInfo.getBucketName(),
+              omKeyInfo.getKeyName());
+      // For OmResponse with failure, this should do nothing. This method is
+      // not called in failure scenario in OM code.
+      keyTable.deleteWithBatch(batchOperation, ozoneDbKey);
```

Comment:
keyTable.deleteWithBatch(batchOperation, ozoneDbKey); seems repeated, this operation also done in addDeletionToBatch.
```java
  protected void addDeletionToBatch(
      OMMetadataManager omMetadataManager,
      BatchOperation batchOperation,
      Table<String, ?> fromTable,
      String keyName, String fullKeyName,
      OmKeyInfo omKeyInfo) throws IOException {

    // For OmResponse with failure, this should do nothing. This method is
    // not called in failure scenario in OM code.
    fromTable.deleteWithBatch(batchOperation, keyName);
    ...
```

Reply:
Thanks for finding this 🙂